test/test_glacier_utils.py

```diff
@@ -0,0 +1,271 @@
+                    'VersionId': 'rOthWbK8m0MFeX9Y_k4E4V69gJu8n0RU',
+                    'IsLatest': True,
+                    'ETag': '"abc123"',
+                    'Size': 1024,
+                    'StorageClass': 'STANDARD'
+                },
+                {
+                    'Key': 'example.txt',
+                    'VersionId': 'Y4D4UNy1m4vLyJ7fIQdSPS1LzZuT8TJG',
+                    'IsLatest': False,
+                    'ETag': '"def456"',
+                    'Size': 2048,
+                    'StorageClass': 'GLACIER'
+                }
+            ],
+            'Name': 'dummy-bucket',
+            'KeyCount': 2,
+        },
+        {
+            'Versions': [
+                {
+                    'Key': 'example.txt',
+                    'VersionId': 'Y4D4UNy1m4vLyJ7fIQdSPS1LzZuT8TJG',
+                    'IsLatest': False,
+                    'ETag': '"def456"',
+                    'Size': 2048,
+                    'StorageClass': 'GLACIER'
+                }
+            ],
+            'Name': 'dummy-bucket',
+            'KeyCount': 1,
+        },
+        {
+            'Versions': [
+                {
+                    'Key': 'example.txt',
+                    'VersionId': 'Y4D4UNy1m4vLyJ7fIQdSPS1LzZuT8TJG',
+                    'IsLatest': False,
+                    'ETag': '"def456"',
+                    'Size': 2048,
+                    'StorageClass': 'GLACIER_IR'
+                }
+            ],
+            'Name': 'dummy-bucket',
+            'KeyCount': 1,
+        },
+        {
+            'Versions': [
+                {
+                    'Key': 'example.txt',
+                    'VersionId': 'rOthWbK8m0MFeX9Y_k4E4V69gJu8n0RU',
+                    'IsLatest': True,
+                    'ETag': '"abc123"',
+                    'Size': 1024,
+                    'StorageClass': 'STANDARD'
+                },
+                {
+                    'Key': 'example.txt',
+                    'VersionId': 'Y4D4UNy1m4vLyJ7fIQdSPS1LzZuT8TJG',
+                    'IsLatest': False,
+                    'ETag': '"def456"',
+                    'Size': 2048,
+                    'StorageClass': 'DEEP_ARCHIVE'
+                }
+            ],
+            'Name': 'dummy-bucket',
+            'KeyCount': 2,
+        }
+    ])
+    def test_glacier_utils_delete_glaciered_versions_exist(self, glacier_utils, response):
+        """ Tests some mocked deletes of dummy list_objects calls for a particular object with a standard
+            and glacier version, should return true """
+        gu = glacier_utils
+        with mock.patch.object(gu.s3, 'list_object_versions',
+                               return_value=response):
+            with mock.patch.object(gu.s3, 'delete_object'):
+                assert gu.delete_glaciered_object_versions('bucket', 'key')
+                assert gu.delete_glaciered_object_versions('bucket', 'key', delete_all_versions=True)
```

Comment:
After doing these calls, you should be testing the calls really occurred. e.g.,

```
            with mock.patch.object(gu.s3, 'delete_object') as mock_delete:
                assert gu.delete_glaciered_object_versions('bucket', 'key')
                assert mock_delete.call_count == 1
                mock_delete.assert_called_with(Bucket=..., Key=..., VersionId=...)  # tests most recent call
                mock_delete.reset_mock()
                assert gu.delete_glaciered_object_versions('bucket', 'key', delete_all_versions=True)
                assert mock_delete.call_count == 2
                mock_delete.assert_has_calls([
                    mock.call(Bucket=..., Key=..., VersionId=...),
                    mock.call(Bucket=..., Key=..., VersionId=...)
                ],
                any_order=True)
```
The `any_order=True` may not be necessary as you've written it but could be if you add parallelism.